doc/circuitikzmanual.tex

```diff
@@ -1774,6 +1774,91 @@ \subsubsection{Transistor paths}\label{sec:transasbip}
 
 
 \subsection{Electronic Tubes}
+
+\small{Contributed by J. op den Brouw.}
+
+\begin{groupdesc}
+	\circuitdesc*{diodetube}{Tube Diode}{} ( anode/90/0.2, cathode/-90/0.2 )
+	\circuitdesc*{triode}{Triode}{} ( anode/90/0.2, cathode/-90/0.2, grid/180/0.2 )
+	\circuitdesc*{tetrode}{Tetrode}{} ( anode/90/0.2, cathode/-90/0.2, grid/190/0.2,screen/170/0.2 )
+	\circuitdesc*{pentode}{Pentode}{} ( anode/90/0.2, cathode/-90/0.2, grid/190/0.2,screen/180/0.2,suppressor/170/0.2 )
+\end{groupdesc}
+
+Note that the \verb|diodetube| is used as component name to avoid clashes with the semiconductor diode.
+Normally, the filament is not drawn. If you want a filament, put the \verb|filament| option in the node description:
+
+\begin{groupdesc}
+	\circuitdesc*{diodetube,filament}{Tube Diode}{} ( anode/90/0.2, filament 1/-135/0.2, filament 2/-45/0.2 )
+\end{groupdesc}
+
+Sometimes, you don't want the cathode to be drawn (but you do want the filament). Use the \verb|nocathode| option in the node description:
+
+\begin{groupdesc}
+	\circuitdesc*{diodetube,filament,nocathode}{Tube Diode}{} ( anode/90/0.2 )
+\end{groupdesc}
+
+If you want a full cathode to be drawn, use the \verb|fullcathode| option in the node description. You can then use the anchors \verb|cathode 1| and \verb|cathode 2|.
+
+\begin{groupdesc}
+	\circuitdesc*{diodetube,fullcathode}{Tube Diode}{} ( anode/90/0.2, cathode 1/-135/0.2, cathode 2/-45/0.2 )
+\end{groupdesc}
+
```

Comment:
```suggestion

\begin{tabular}{l | l | l}
  Key & Default value & Description\\
  \hline
  \verb|tubes/width/.initial| & \verb|1|                    & relative width               \\
  \verb|tubes/height/.initial| & \verb|1.4|                 & relative height              \\
  \verb|tubes/tube radius/.initial| & \verb|0.40|           & radius of tube circle        \\
  \verb|tubes/anode distance/.initial| & \verb|0.40|        & distance from center         \\
  \verb|tubes/anode width/.initial| & \verb|0.40|           & width of an anode/plate      \\
  \verb|tubes/grid protrusion/.initial| & \verb|0.25|       & distance from center         \\
  \verb|tubes/grid dashes/.initial| & \verb|5|              & number of grid dashes        \\
  \verb|tubes/grid separation/.initial| & \verb|0.2|        & separation between grids     \\
  \verb|tubes/grid shift/.initial| & \verb|0.0|             & y shift of grids from center \\
  \verb|tubes/cathode distance/.initial| & \verb|0.40|      & distance from grid           \\
  \verb|tubes/cathode width/.initial| & \verb|0.40|         & width of a cathode           \\
  \verb|tubes/cathode corners/.initial| & \verb|0.06|       & corners of the cathode wire  \\
  \verb|tubes/cathode right extend/.initial| & \verb|0.075| & extension at the right side  \\
  \verb|tubes/filament distance/.initial| & \verb|0.1|      & distance from cathode        \\
  \verb|tubes/filament angle/.initial| & \verb|15|          & angle from the centerpoint   \\
\end{tabular}
```

Reply:
Don't merge this yet please, I have plans for it :)

Reply:
OK. I'll wait (and see)...

Reply:
My plans were a little too hard 😆 ... feel free to accept the suggestion if you feel it makes sense

Reply:
@Rmano, this is something I'd like to do on all pull requests which add parts going forward; if you get around to that API refactor, we can do it for the core parts there as well.

Reply:
@jesseopdenbrouw this should be good to merge.

Reply:
@asinghvi17 yes, the tabular form is nice, although it will stretch even more the page-breaking mechanism (we probably should use a kind of breakable table, like longtable or similar). Another way of listing options is the one used by the TikZ manual, but to be honest I think that a more compact way like this is better for circuitikz (where most parameters are quite easy to understand). 

@jesseopdenbrouw --- if you are fine please merge the changes and then I'll merge the whole PR.